packages/next-dashboard/src/components/layout/Main.js

```diff
@@ -61,75 +62,86 @@ function DashboardLayout({
       window.removeEventListener('resize', noTransition);
     };
   }, []);
+  const context = useContext(DashboardContext);
+  if (!context) {
+    throw new TypeError('Dashboard Layout needs to be in a Dashboard Context');
+  }
+
+  const {
+    getState,
+    theme: { class: themeClass },
+    modalActive,
+  } = context;
+
+  const sidebarActive = getState<boolean>('sidebarActive', false);
+  const sidebarCompact = getState<boolean>('sidebarCompact', false);
+
+  const toggleSidebarActive = () => {
+    context.setState<boolean>('sidebarActive', !sidebarActive);
+  };
+
+  const [scrollOffset, setScrollOffset] = useState<number | null>(null);
+  const [savedScrollOffset, setSavedScrollOffset] = useState<number | null>(
+    null,
+  );
+
+  // When a modal is shown, we want to save the top offset and set position fixed to disable scrolling
+  useEffect(() => {
+    if (scrollOffset === null && !modalActive) return;
+    if (modalActive) setScrollOffset(window.pageYOffset);
+    else {
+      setSavedScrollOffset(scrollOffset);
+      setScrollOffset(null);
+    }
```

Comment:
Minor: stylistic, but I try to be consistent with braces on either no or all branches of an if-else chain. So, I'd put braces around the "then" case here and put the `setScrollOffset(window.pageYOffset);` on its own line

Reply:
Yeah, the "setSavedScrollOffset" came afterwards and forgot to add the brackets to the leading if